docs/builtins.md

```diff
@@ -0,0 +1,730 @@
+string. The format string is the first argument, and it
+contains _specifiers_. Specifiers are placeholder that start with a
+percent sign `%`. They tell the `printf` function how and where to
+print the remaining arguments inside the format string. The rest of the
+format string is printed to the output area without changes. 
+
+Here are some valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| `%v`      | the argument in its default format  |
+| `%q`      | a double-quoted string |
+| `%%`      | a percent sign `%`  |
+
+#### Example
+
+```evy
+printf "%%v string: %v\n" "abc"
+printf "%%v bool: %v\n" true
+printf "%%v num: %v\n" 1
+print
+printf "%%q string: %q\n" "abc"
+printf "%%v num with decimals: %v\n" 1.234
+print
+printf "%%v array: %v\n" [1 2 3]
+printf "%%v map: %v\n" {a:1 b:2}
+```
+
+Output
+```
+%v string: abc
+%v bool: true
+%v num: 1
+
+%q string: "abc"
+%v num with decimals: 1.234
+
+%v array: [1 2 3]
+%v map: {a:1 b:2}
+```
+
+#### Reference 
+
+    printf format:string a:any...
+
+The `printf` function prints its arguments to the output area according
+to the _format_ string that is represented by the first argument.
+The _specifiers_ that start with `%` and are contained in the format
+string are replaced by the remaining arguments in the given order. For
+example, the following code `printf "first: %s, second: %s" "A" "B"`
+prints `first: A, second: B`.
+
+Full list of valid specifiers in Evy:
+
+| Specifier | Description |
+| --------- | ------------- |
+| %v        | the argument in a default format  |
+| %%        | a literal percent sign `%`; consumes no value  |
+| %t        | the word `true` or `false` |
+| %f        | decimal point (floating-point) number, e.g. 123.456000 |
+| %e        | scientific notation, e.g. -1.234456e+78 |
+| %s        | string value |
+| %q        | a double-quoted string |
+
+If the arguments for the `%s`, `%q`, `%f`, `%e`, and `%t` specifiers do
+not match the required type, a runtime error will occur.
+
+The _width_ and _precision_ of a floating-point number can be specified
+with the `%f` and `%v` format specifiers.
+
+- Width is the number of characters that will be used to print the
+  number. If the width is not specified, it will be calculated based on
+  the size of the number. It can be useful for padding and aligned
+  output.
+- Precision is the number of decimal places that will be displayed. If
+  the precision is not specified, it will be set to 6. Here is a table
+  that shows the different ways to specify the width and precision of a
+  floating-point number:
+
+
+| Verb    | Description |
+| ------- | ------------- |
+| %f      | default width, default precision |
+| %7f     | width 7, default precision |
+| %.2f    | default width, precision 2 |
+| %7.2f   | width 7, precision 2 |
+| %7.f    | width 7, precision 0 |
+
+
+The `-` flag tells `printf` to pad the output with spaces on the right,
```

Comment:
I think "flag" is the wrong word to use here. I would perhaps say:

If the width/precision is preceded by a `-`, the value is padded with spaces on the
right rather than the left. If it is preceded by a `0`, the value is padded with leading
zeros rather than spaces.

Reply:
done.